bundles/EcommerceFrameworkBundle/Model/MockProduct.php

```diff
@@ -0,0 +1,115 @@
+<?php
+
+/**
+ * Pimcore
+ *
+ * This source file is available under two different licenses:
+ * - GNU General Public License version 3 (GPLv3)
+ * - Pimcore Commercial License (PCL)
+ * Full copyright and license information is available in
+ * LICENSE.md which is distributed with this source code.
+ *
+ *  @copyright  Copyright (c) Pimcore GmbH (http://www.pimcore.org)
+ *  @license    http://www.pimcore.org/license     GPLv3 and PCL
+ */
+
+namespace Pimcore\Bundle\EcommerceFrameworkBundle\Model;
+
+use Pimcore\Bundle\EcommerceFrameworkBundle\AvailabilitySystem\AvailabilityInterface;
+use Pimcore\Bundle\EcommerceFrameworkBundle\AvailabilitySystem\AvailabilitySystemInterface;
+use Pimcore\Bundle\EcommerceFrameworkBundle\Factory;
+use Pimcore\Bundle\EcommerceFrameworkBundle\PriceSystem\PriceInfoInterface;
+use Pimcore\Bundle\EcommerceFrameworkBundle\PriceSystem\PriceInterface;
+use Pimcore\Bundle\EcommerceFrameworkBundle\PriceSystem\PriceSystemInterface;
+use Pimcore\Model\DataObject\Concrete;
+
+/**
+ * Mock Product class which should be used as a product when actual product is not available in the System.
+ */
+class MockProduct extends Concrete implements ProductInterface, IndexableInterface, CheckoutableInterface
+{
+    public function getAvailabilitySystemName(): ?string
```

Comment:
I would return null where ever possible. 
so also with `getAvailabilitySystemName`, `getPriceSystemImplementation`, `getAvailabilitySystemImplementation`, `getOSPriceInfo`, `getOSAvailabilityInfo`, `getPriceSystemName`

Reply:
It is not possible to return null for these methods due to incompatibility between CartItemInterface & CheckoutableInterface. I have created an issue for Pimcore 11 https://github.com/pimcore/pimcore/issues/9405